scipy/interpolate/_polyint.py

```diff
@@ -677,8 +742,91 @@ def _evaluate(self, x):
                 p[r[:-1]] = self.yi[r[-1]]
         return p
 
+    def derivative(self, x, der=1):
```

Comment:
```suggestion
    def derivative(self, x, *, der=1):
```

Reply:
Here `derivative` is providing a concrete implementation of the (pre-existing) abstract method of the base class `_Interpolator1DWithDerivatives` which has signature `derivative(self, x, der=1)`, so shouldn't it match?

Reply:
I would prefer to see this match the signature of the base class